webxr-api/input.rs

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::Input;
+use crate::Native;
+
+use euclid::TypedRigidTransform3D;
+
+#[derive(Copy, Clone, Debug)]
+#[cfg_attr(feature = "ipc", derive(serde::Serialize, serde::Deserialize))]
+pub enum Handedness {
+    None,
+    Left,
+    Right,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[cfg_attr(feature = "ipc", derive(serde::Serialize, serde::Deserialize))]
+pub enum TargetRayMode {
+    Gaze,
+    TrackedPointer,
+    Screen,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[cfg_attr(feature = "ipc", derive(serde::Serialize, serde::Deserialize))]
+pub struct InputInfo {
+    pub handedness: Handedness,
+    pub target_ray_mode: TargetRayMode,
+    pub id: u32,
+}
+
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "ipc", derive(serde::Serialize, serde::Deserialize))]
+pub struct InputFrame {
```

Comment:
We're going to want more of these as time goes no, so make this an `enum InputEvent`?

Reply:
No, events should not be handled here

Reply:
OKI, but if this is just for pose info then just using `TypedTransform3D<f32, Input, Native>` seems simplest.